module/scripts/index/commons-source-ui.js

```diff
@@ -18,16 +18,35 @@ Refine.CommonsSourceUI = function(controller) {
 
     var self = this;
 
+    var cmtitle = $.trim(self._elmts.categoryInput[0].value);
+    var inputContainer = $('<div></div>').appendTo(cmtitle);
+    var input = $('<input></input>').appendTo(inputContainer);
+
+    var endpoint = "https://commons.wikimedia.org/w/api.php?action=opensearch&format=json&formatversion=2&search="
+    + cmtitle + "&namespace=14&limit=10";
```

Comment:
The endpoint should just be the base URL of the API - the user input will be added to the query later on (every time the user input changes)